src/sentry/runner/commands/configoptions.py

```diff
@@ -0,0 +1,267 @@
+import re
+
+import click
+
+from sentry.runner.decorators import configuration
+
+tracked = [
+    re.compile("sentry:test_key.+"),
+]
+
+verbose = False
+
+
+def create_key_value_generator(data: str, newline_separator: str, kv_separator: str):
+    return (line.split(kv_separator) for line in data.split(newline_separator) if line)
+
+
+@click.group()
+def configoptions():
+    "Manages Sentry options."
+
+
+@configoptions.command()
+@configuration
+def list():
+    "Fetches all options."
+    from sentry import options
+
+    for opt in options.all():
+        # click.echo(f"{opt.name} ({opt.type}) = {options.get(opt.name)}")
+        # click.echo(type(opt))
+        if can_change(opt.name):
+            click.echo(f"{opt.name}: {options.get(opt.name)}")
+
+
+@configoptions.command()
+@click.argument("filename", required=True)
+@click.option(
+    "--dryrun",
+    is_flag=True,
+    default=False,
+    required=False,
+    help="Output exactly what changes would be made and in which order.",
+)
+@configuration
+def patch(filename: str, dryrun: bool):
+    "Updates, gets, and deletes options that are each subsectioned in the given file."
+    import yaml
+
+    if dryrun:
+        click.echo("Dryrun flag on. ")
+    with open(filename) as stream:
+        configmap_data = yaml.safe_load(stream)
+        data = configmap_data.get("data", {}).get("options-patch.yaml", "")
+
+        keysToFetch = data.get("fetch", {})
+        keysToFetch = create_key_value_generator(keysToFetch, "\n", ": ")
+        for key in keysToFetch:
+            _get(key, dryrun)
+
+        keysToUpdate = data.get("update", {})
+        keysToUpdate = create_key_value_generator(keysToUpdate, "\n", ": ")
```

Comment:
I will resurface my previous concern as the comment got orphaned and is not visible anymore.

Please let's not store the options as string that has to be deserialized. This has a number of disadvantages vs storing the plain valid yaml:
- We have constraints in what is an option today: simple values, lists, sets, mapping. Everything is supposed to be simply serializable so we are not supposed to have the problem that requires serialization.
- escaping. You will have to escape values to make them valid json/yaml when they are in a string. This will make mistakes easier.
- linting of the options file will be harder because the main yaml file will always be valid yaml whether the content of the string is properly formatted or not.
- changing back to yaml afterwards will be complex.